### ISSUE TITLE
Change abjad.MetronomeMark formatting

### ISSUE THREAD

**Issue by trevorbaca** [Abjad/abjad#172]
_From [trevorb...@gmail.com](https://code.google.com/u/118334091018461915321/) on October 23, 2012 11:38:32_
ONE:
```
\new Score <<
    \new Staff {
        \times 2/3 {
            \tempo 4=48
            c'8
            d'8
            e'8
        }
    }
>>
```
TWO:
```
\new Score <<
    \tempo 4=48
    \new Staff {
        \times 2/3 {
            c'8
            d'8
            e'8
        }
    }
>>
```

The ticket asks for the formatting code to always format like #1 instead of #2.

_Original issue: http://code.google.com/p/abjad/issues/detail?id=171_

**Comment by josiah-wolf-oberholtzer** [comment]
Wasn't there a problem with this where putting the tempo indication immediately inside the score block caused LilyPond to implicitly generate an empty staff?

**Comment by trevorbaca** [comment]
@josiah-wolf-oberholtzer Yeah, I think so.

**Comment by trevorbaca** [comment]
Still valid in May 2017:
```
staff = Staff(r"\times 2/3 { c'8 d'8 e'8 }")
score = Score([staff])
tempo = Tempo(Duration(1, 4), 48)
attach(tempo, staff)
f(score)

\new Score <<
    \new Staff {
        \tempo 4=48
        \times 2/3 {
            c'8
            d'8
            e'8
        }
    }
>>
```

**Comment by josiah-wolf-oberholtzer** [comment]
Formatting like the following will cause LilyPond to implicitly generate an extra voice context:

```
\new Score <<
    \tempo 4=48
    \new Staff {
        \times 2/3 {
            c'8
            d'8
            e'8
        }
    }
>>
```

![test](https://cloud.githubusercontent.com/assets/5505674/25822602/f1164ac6-33ed-11e7-8b61-657a8e37d5b4.png)

These two formulations do not create an extra context:

```
\new Score <<
    \new Staff {
        \tempo 4=48
        \times 2/3 {
            c'8
            d'8
            e'8
        }
    }
>>
```

...and...

```
\new Score <<
    \new Staff {
        \times 2/3 {
            \tempo 4=48
            c'8
            d'8
            e'8
        }
    }
>>
```

**Comment by trevorbaca** [comment]
Great point. (Also love the embedded notation.)

Seems like the obvious solution is always to only ever format immediately before a leaf.

**Comment by josiah-wolf-oberholtzer** [comment]
I agree. Formatting before the leaf both gives the desired graphical output in LilyPond and also guarantees the simplest formatting logic of the various possible tempo-formatting locations.

I propose you close out this issue.

**Comment by josiah-wolf-oberholtzer** [comment]
Hmm. I'm leaning towards indicators only attaching to leaves, although there's definitely some disconnect between our model and LilyPond's there (tempo events are stand-alone stream events in LilyPond, unlike articulations). But it ultimately makes things simpler if they just attach to leaves, no matter what. That's for sure.

**Comment by trevorbaca** [comment]
Totally. It's 100% a simplicity thing. It's been a couple of releases now that spanners have been restricted to attaching only to leaves, and it seems to work great.

Two other pretty huge benefits to an only-to-leaves restriction for indicator attachment:

1. The threeway (visual) mapping between PDF-ly-datastructure becomes much more isomorphic with the restriction in place. (See a clef somewhere in a PDF? Currently it might be attached to any number of components and might appear in any number of different slots in the corresponding ly file. With an only-to-leaves restriction in place the clef is attached to the following leaf unambiguously.)

2. Use of `inspect()` finally becomes massively simpler in an important set of circumstances. Currently calling `inspect(leaf).has_indicator(clef)` sometimes yields conceptually surprising "false negatives" when there's a clef not attached to the leaf in question but to a container in the leaf's parentage. Especially when the leaf is sitting at index 0 in a container this can be really conceptually difficult to work with.

Those two changes should benefit all users at some point or other.

I suppose there's a third benefit that's developer-oriented but still pretty important: the implementation of almost every mutate method will become far simpler and far easier to maintain in the longterm.

All just observations for the moment; no changes coming anytime too soon.

**Comment by josiah-wolf-oberholtzer** [comment]
Sounds good to me. I'm comfortable with this simplification.

**Comment by trevorbaca** [comment]
Depends on #824.

Finish #824 and this problem will disappear.

**Comment by trevorbaca** [comment]
Duplicate of #824.

**Comment by trevorbaca** [comment]
Indicators restricted to leaves in 2.21.
